query-coordinator/src/main/scala/com/socrata/querycoordinator/fusion/CompoundTypeFuser.scala

```diff
@@ -74,7 +79,41 @@ class CompoundTypeFuser(fuseBase: Map[String, String]) extends SoQLRewrite with
     }
   }
 
-  def rewrite(select: Select): Select = {
+  def rewrite(parsedStmts: Seq[Select],
+              columnIdMapping: Map[ColumnName, String],
+              schema: Map[String, SoQLType]): Seq[Select] = {
+
+    // expand "select *"
+    val firstStmt = parsedStmts.head
+
+    val baseCtx = new UntypedDatasetContext {
+      override val columns: OrderedSet[ColumnName] = {
+        // exclude non-existing columns in the schema
+        val existedColumns = columnIdMapping.filter { case (k, v) => schema.contains(v) }
+        OrderedSet(existedColumns.keysIterator.toSeq: _*)
+      }
+    }
+
+    val expandedStmts = parsedStmts.foldLeft((Seq.empty[Select], toAnalysisContext(baseCtx))) { (acc, select) =>
+      val (selects, ctx) = acc
+      val expandedSelection = AliasAnalysis.expandSelection(select.selection)(ctx)
+      val expandedStmt = select.copy(selection = Selection(None, Seq.empty, expandedSelection))
+      val columnNames = expandedStmt.selection.expressions.map { se =>
+        se.name.map(_._1).getOrElse(ColumnName(se.expression.toString.replaceAllLiterally("`", "")))
```

Comment:
Column name is column alias or field name.  I don't remember why it tries column alias.  That code isn't changed.  It is moved from QueryParser to Fuser.  The fuser is no longer shunt for soqls that have joins.